lib/services/workflow-api-service.js

```diff
@@ -237,7 +237,39 @@ function workflowApiServiceFactory(
     WorkflowApiService.prototype.getWorkflowsByNodeId = function(id, query) {
         var nodeId = ({node: id});
         var mergedQuery = _.merge({}, nodeId, query);
-        return waterline.graphobjects.find(mergedQuery);
+        return waterline.graphobjects.find(mergedQuery).then(
+            function sanitize(workflow) {
+                var sanitizedWorkflow = workflow;
+                var walked = [];
+                var stack = [ { workflow : workflow , stack : '' } ];
+                while ( stack.length > 0 ) {
+                    var item = stack.pop();
+                    var workflow  = item.workflow;
+                    for ( var property in workflow ) {
+                        if ( workflow.hasOwnProperty( property )) {
+                            if ( typeof workflow[ property ] == "object" ) {
+                                var alreadyFound = false;
+                                for ( var i = 0; i < walked.length; i++ ) {
+                                    if ( walked[i] === workflow[ property ] ) {
+                                        alreadyFound = true;
+                                        break;
+                                    }
+                                }
+                                if ( !alreadyFound ) {
+                                    walked.push( workflow[ property ]);
+                                    stack.push( { workflow : workflow[ property ], stack: item.stack + '.' + property });
+                                }
+                            } else {
+                                if ( property.match( /password/i )) {
+                                    workflow[property] = '*** REDACTED ***';
```

Comment:
@jlongever So I had this discussion with a few people, and I'm of the camp that by redacting the field it insures there is no confusion in testing... if this output is used to verify the workflow has all the expected keys.